coordinator/src/routes.rs

```diff
@@ -604,3 +617,132 @@ pub async fn version() -> Result<Json<Version>, AppError> {
         branch: env!("BRANCH_NAME").to_string(),
     }))
 }
+
+pub async fn collaborative_revert_confirm(
+    State(state): State<Arc<AppState>>,
+    revert_params: Json<CollaborativeRevertData>,
+) -> Result<Json<String>, AppError> {
+    let mut conn = state.pool.clone().get().map_err(|error| {
+        AppError::InternalServerError(format!("Could not acquire db lock {error:#}"))
+    })?;
+
+    let channel_id_string = revert_params.channel_id.clone();
+    let channel_id = hex::decode(channel_id_string.clone())
+        .map_err(|err| AppError::BadRequest(err.to_string()))?;
+    let channel_id: [u8; 32] = channel_id
+        .try_into()
+        .map_err(|_| AppError::BadRequest("Provided channel ID was invalid".to_string()))?;
+
+    tracing::info!(
+        channel_id = channel_id_string,
+        "Confirming collaborative channel revert"
+    );
+    let inner_node = state.node.inner.clone();
+
+    // TODO: check if provided amounts are as expected
+    if !revert_params
+        .transaction
+        .output
+        .iter()
+        .any(|output| inner_node.wallet().is_mine(&output.script_pubkey).is_ok())
+    {
+        let error_message = "Invalid request: no address for coordinator provided".to_string();
+        tracing::error!(error_message);
+        return Err(AppError::BadRequest(error_message));
+    }
+
+    let sub_channels = inner_node.list_dlc_channels().map_err(|e| {
+        AppError::InternalServerError(format!("Failed to list DLC channels: {e:#}"))
+    })?;
+    let subchannel = sub_channels
+        .iter()
+        .find(|c| c.channel_id == channel_id)
+        .context("Could not find provided channel")
+        .map_err(|e| AppError::BadRequest(format!("Channel not found: {e:#}")))?;
+
+    let channel_manager = inner_node.channel_manager.clone();
+
+    let mut own_sig = None;
+
+    let mut revert_transaction = revert_params.transaction.clone();
+
+    let position = Position::get_position_by_channel_id(&mut conn, channel_id_string.clone())
```

Comment:
🔧 if you would go for the `user_channel_id`, you wouldn't have to add a new database query to fetch the channel by `channel_id`

Reply:
Even better, why don't you simply provide already the counterparty pubkey from the subchannel, instead of taken the d-tour over the `channel_id`?

Reply:
good point. Thanks.

Reply:
[`c505c6d` (#1478)](https://github.com/get10101/10101/pull/1478/commits/c505c6dec395e48931fe505300efe4d440991a32)